src/messaging/messaging.ios.ts

```diff
@@ -551,7 +547,13 @@ class UNUserNotificationCenterDelegateImpl extends NSObject implements UNUserNot
     this.callback(notification);
   }
 
-  public userNotificationCenterDidReceiveNotificationResponseWithCompletionHandler(center: UNUserNotificationCenter, response: UNNotificationResponse, completionHandler: () => void): void {
+  public userNotificationCenterDidReceiveNotificationResponseWithCompletionHandler(center: UNUserNotificationCenter, response: UNNotificationResponse, completionHandler: () => void, next: () => void): void {
+    const userInfo = response.notification.request.content.userInfo;
+    const userInfoJSON = firebaseUtils.toJsObject(userInfo);
+    if (!userInfoJSON["gcm.message_id"]) { // not a firebase message!
+      next();
+      return;
+    }
     // let's ignore "dismiss" actions
```

Comment:
This will probably not work for non-firebase messages. I'm using azure notification hubs, and it doesn't have a `gcm.message_id` property. There is a `data.aps` property that might be good to check here as well.

Reply:
Actually here is a better way to check if the push is from remote or not (since it doesn't rely on any JSON properties in the payload):
https://stackoverflow.com/a/48717664/5272836